rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/SchedulerStateRest.java

```diff
@@ -1161,6 +1165,32 @@ public String getJobHtml(@HeaderParam("sessionid") String sessionId, @PathParam(
         }
     }
 
+    @Override
+    @GET
+    @Path("jobs/{jobid}/xml")
+    @Produces("application/xml")
+    public InputStream getJobContent(@HeaderParam("sessionid") String sessionId, @PathParam("jobid") String jobId)
+            throws NotConnectedRestException, UnknownJobRestException, PermissionRestException,
+            SubmissionClosedRestException, JobCreationRestException {
+        try {
+            Scheduler s = checkAccess(sessionId, "/scheduler/jobs/" + jobId + "/xml");
+            final Job jobContent = s.getJobContent(JobIdImpl.makeJobId(jobId));
+            return new Job2XMLTransformer().jobToxml((TaskFlowJob) jobContent);
```

Comment:
you need to be protective in case jobContent is null

Reply:
did it